contrib_bots/lib/yoda_bot.py

```diff
@@ -0,0 +1,129 @@
+# See readme-yoda-bot.md for instructions on running this code.
+from __future__ import print_function
+import unirest
+import os
+import logging
+import ssl
+
+
+class ServerError(ssl.SSLError):
+    '''raise this when there is an error with the server'''
+
+
+class ApiKeyError(Exception):
+    '''raise this when there is an error with the Mashape Api Key'''
+
+
+class YodaSpeakHandler(object):
+    '''
+    This bot will allow users to translate a sentence into 'Yoda speak'.
+    It looks for messages starting with '@yoda'.
+    '''
+
+    def usage(self):
+        return '''
+            This bot will allow users to translate a sentence into
+            'Yoda speak'.
+            Users should preface messages with '@yoda'.
+
+            Before running this, make sure to get a Mashape Api token.
+            Instructions are in the 'readme-yoda-bot.md' file.
+            Store it in the 'yoda_api_key.txt' file.
+            The 'yoda_api_key.txt' file should be located at '~/yoda_api_key.txt'.
+            Example input:
+            @yoda You will learn how to speak like me someday.
+            '''
+
+    def triage_message(self, message):
+        # return True iff we want to (possibly) response to this message
+
+        original_content = message['content']
+
+        is_sentence = (original_content.startswith('@yoda'))
+
+        return is_sentence
+
+    def handle_message(self, message, client, state_handler):
+        original_content = message['content']
+        stream = message['display_recipient']
+        subject = message['subject']
+
+        # this handles the message if its starts with @yoda
+        if original_content.startswith('@yoda'):
+            handle_input(client, original_content, stream, subject)
+
+handler_class = YodaSpeakHandler
+
+
+def send_to_yoda_api(sentence, api_key):
+    # function for sending sentence to api
+
+    try:
+        response = unirest.get("https://yoda.p.mashape.com/yoda?sentence=" + sentence,
+                               headers={
+                                   "X-Mashape-Key": api_key,
+                                   "Accept": "text/plain"
+                               }
+                               )
+
+        if response.code == 200:
+            return response.body
+        if response.code == 403:
+            raise ApiKeyError
+        else:
+            error_message = response.body['message']
+            logging.error(error_message)
+            error_code = response.code
+            error_message = error_message + 'Error code: ' + error_code + \
+                '. Did you follow the instructions in the `readme-yoda-bot.md` file?'
+            return error_message
+
+    except ssl.SSLError:
```

Comment:
No need to catch this here. In fact, removing exception handing from this function will not be a bad idea since you are catching exceptions in `handle_input`.